File: remove_bg.py
import sys
try:
    from PIL import Image, ImageFilter
except ImportError:
    import subprocess
    subprocess.check_call([sys.executable, "-m", "pip", "install", "Pillow"])
    from PIL import Image, ImageFilter

def remove_background(input_path, output_path, tolerance=40):
    img = Image.open(input_path)
    img = img.convert("RGBA")
    
    # Assuming top-left pixel is the background color
    bg_color = img.getpixel((0, 0))
    
    data = img.getdata()
    new_data = []
    
    for item in data:
        # Check Manhattan distance from bg_color
        diff = sum(abs(item[i] - bg_color[i]) for i in range(3))
        if diff < tolerance:
            new_data.append((item[0], item[1], item[2], 0))
        else:
            # Soften edges slightly by checking distance
            if diff < tolerance + 30:
                alpha = int(255 * ((diff - tolerance) / 30))
                new_data.append((item[0], item[1], item[2], alpha))
            else:
                new_data.append(item)
            
    img.putdata(new_data)
    img.save(output_path, "PNG")
    print("Successfully converted logo to PNG with transparent background.")

remove_background("public/images/logo.jpeg", "public/images/logo.png")

File: remove_bg.js
const Jimp = require("jimp");

async function removeBackground() {
  try {
    const image = await Jimp.read("public/images/logo.jpeg");
    
    // Get the color of the top-left pixel
    const bgPixel = image.getPixelColor(0, 0);
    const bgRgba = Jimp.intToRGBA(bgPixel);

    image.scan(0, 0, image.bitmap.width, image.bitmap.height, function (x, y, idx) {
      const r = this.bitmap.data[idx + 0];
      const g = this.bitmap.data[idx + 1];
      const b = this.bitmap.data[idx + 2];
      
      const dist = Math.abs(r - bgRgba.r) + Math.abs(g - bgRgba.g) + Math.abs(b - bgRgba.b);
      
      if (dist < 40) {
        this.bitmap.data[idx + 3] = 0;
      } else if (dist < 90) {
        this.bitmap.data[idx + 3] = Math.floor(255 * ((dist - 40) / 50));
      }
    });

    await image.writeAsync("public/images/logo.png");
    console.log("Success");
  } catch (err) {
    console.error(err);
  }
}

removeBackground();
